### ISSUE TITLE
Tagoverflow: dom nesting and tooltip accessibility

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/ibm-products#5736]
### Package

Carbon for IBM Products

### Description

i noted that the [TagOverflow](https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-tag-overflow-tagoverflow--tags-with-truncation&globals=viewport:basic) story is displaying a invalid dom error. `button cannot be a descendent of button`. this notification appears to only pop up locally.

the issue stems from `Tooltip` being wrapped around the `Tag` components. because tooltip is a button and tag's have an `onClick` mechanism they are rendered as buttons.

in addition to this in and of itself being an accessibility issue, this is also tied to our ongoing efforts to remove unnecessary usage of tooltips and `title` attributes. my suggestion is to remove the tooltip entirely. since carbon tags already utilize `text-overflow` i think what we should do is allow the ellipsis to show as usual and then adjust our popover detection mechanism to check for truncated values and display them in full in the modal. i believe that because this offers a way for users who need screen readers a way to access the full value that this fix would be accessible.

### Component(s) impacted

[TagOverflow](https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-tag-overflow-tagoverflow--tags-with-truncation&globals=viewport:basic)

### Browser

Chrome

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

latest

### Suggested Severity

None

### Product/offering

c4p

### CodeSandbox or Stackblitz example

NA

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues
